clients/src/main/java/org/apache/kafka/clients/consumer/internals/DefaultBackgroundThread.java

```diff
@@ -50,57 +49,83 @@ public class DefaultBackgroundThread extends KafkaThread {
     private final Logger log;
     private final BlockingQueue<ApplicationEvent> applicationEventQueue;
     private final BlockingQueue<BackgroundEvent> backgroundEventQueue;
-    private final ConsumerNetworkClient networkClient;
-    private final SubscriptionState subscriptions;
     private final ConsumerMetadata metadata;
-    private final Metrics metrics;
     private final ConsumerConfig config;
+    // empty if groupId is null
+    private final Optional<CoordinatorRequestManager> coordinatorManager;
```

Comment:
Instead of using `Optional`, could we use a NOOP implementation?

Reply:
Let's reconsider this later? I'm not sure if that would simplify the code.